ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -150,7 +150,10 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
   protected fun addNextButton() =
     addButton(ButtonAction.NEXT)
       .setOnClickListener { handleNext() }
-      .setOnValueChanged { button, value -> button.enableIfTrue(value.isNotNullOrEmpty()) }
+      .setOnValueChanged { button, value ->
+        button.enableIfTrue(value.isNotNullOrEmpty())
+        button.applyDoneState(testLastPosition(value))
```

Comment:
We add the next button in `CaptureLocationTask` and `DropPinTask` as well. This logic should be kept in some shared location.

Reply:
Ah, good catch! Done. I consolidated some of into `TaskButton` but in this case, since it's an overridden method with its own custom action button logic, the best I could do was just add the new callback there.

Technically, we only needed to do this for the MultipleChoice tasks since that's the only way conditional tasks can be triggered, but maybe in the far far future, we'll have conditions based on geographical location or something :)

I tested it with the CaptureLocation as the last task and everything seems to work fine there.